src/mmrelay/matrix/auth.py

```
@@ -0,0 +1,386 @@
+                facade.logger.error(
+                    "Please disable E2EE in your configuration or use a Linux/macOS system for E2EE support."
+                )
+                e2ee_enabled = False
+            else:
+                try:
+                    importlib.import_module("olm")
+
+                    try:
+                        nio_crypto = importlib.import_module("nio.crypto")
+                        if not hasattr(nio_crypto, "OlmDevice"):
+                            raise ImportError("nio.crypto.OlmDevice is unavailable")
```

Ruff (0.15.9)

[warning] Abstract `raise` to an inner function

(TRY301)